packages/ws/src/ws/WebSocketShard.ts

```diff
@@ -259,6 +259,9 @@ export class WebSocketShard extends AsyncEventEmitter<WebSocketShardEventsMap> {
 		this.#status = WebSocketShardStatus.Idle;
 
 		if (options.recover !== undefined) {
+			// There's cases (like no internet connection) where we immediately fail to connect,
+			// causing a very fast and draining reconnection loop.
+			await sleep(500);
```

Comment:
I'd say let's add a debug message for this so people aren't left in the dark, but it's optional from my pov

Reply:
yeah not fond, esp. given it's consistent and not all that long.